domain/src/main/java/life/qbic/domain/events/DomainEventProducer.java

```diff
@@ -0,0 +1,60 @@
+package life.qbic.domain.events;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * <b><class short description - 1 Line!></b>
+ *
+ * <p><More detailed description - When to use, what it solves, etc.></p>
+ *
+ * @since <version tag>
+ */
+public class DomainEventProducer {
+
+  private static final ThreadLocal<List> subscribers = new ThreadLocal<>();
+
+  private static final ThreadLocal<Boolean> publishing = ThreadLocal.withInitial(
+      () -> Boolean.FALSE);
```

Comment:
what does this do?

Reply:
https://www.baeldung.com/java-threadlocal

We basically want to store the current state of the DomainEvent producer in the local thread (which is the user session). It is a flag, that prevents from subscribing when the DomainEventProducer is currently publishing.